rotkehlchen/chain/ethereum/interfaces/balances.py

```diff
@@ -34,19 +41,100 @@ def __init__(
             self,
             database: 'DBHandler',
             evm_inquirer: 'EvmNodeInquirer',
+            chain: 'SupportedBlockchain',
```

Comment:
This should not be any chain. It shoudl only be the subset of chains that are supported in `ProtocolWithBalance`, mainnet and optimism. What's more why use `SupportedBlockchain`? This is only for EvmChain IDs. So this shoudl be chain ID, no? That way you can use the `Location.from_chain_id()` function that already exists and not have to add a new function in `SupportedBlockchain.to_location()`